iptables/iptables.go

```diff
@@ -89,19 +90,39 @@ type Stat struct {
 	Options     string     `json:"options"`
 }
 
-// New creates a new IPTables.
-// For backwards compatibility, this always uses IPv4, i.e. "iptables".
-func New() (*IPTables, error) {
-	return NewWithProtocol(ProtocolIPv4)
+type option func(*IPTables)
+
+func IPFamily(proto Protocol) option {
+	return func(ipt *IPTables) {
+		ipt.proto = proto
+	}
 }
 
-// New creates a new IPTables for the given proto.
-// The proto will determine which command is used, either "iptables" or "ip6tables".
-func NewWithProtocol(proto Protocol) (*IPTables, error) {
-	path, err := exec.LookPath(getIptablesCommand(proto))
+func Timeout(timeout int) option {
+	return func(ipt *IPTables) {
+		ipt.timeout = timeout
+	}
+}
+
+// New creates a new IPTables.
+// For backwards compatibility, by default always uses IPv4, i.e. "iptables".
+func New(opts ...option) (*IPTables, error) {
```

Comment:
Technically this could still be a breaking change if someone was depending on the type of `New` to pass it, but that seems unlikely, and it would be relatively easy to fix I think.

Reply:
If we want to support all the options in iptables in the long term, wait interval comes to mind, I think is the more sustainable way.

I didn't like to add a NewWithProtocolTimeout() constructor, but I can change this to NewWithOptions() though

Reply:
to be clear this LGTM to me already, but I'm not sure how this repo is maintained

Reply:
Hey, I have merge rights, and do releases periodically. Anyways.

Can you update the docblock here to show how options work?